packages/server/src/restaurants/dishes/domain/CreateDishHandler.ts

```diff
@@ -24,16 +24,21 @@ interface CreateDishRequest {
 }
 
 @Injectable()
-class CreateDishHandler implements Handler<CreateDishRequest, Dish> {
+class CreateDishHandler implements Handler<CreateDishRequest, Dish | null> {
   constructor(
     @InjectModel(Dish.name) private dishModel: Model<DishDocument>,
     @InjectModel(Restaurant.name) private restaurantModel: Model<RestaurantDocument>,
   ) {}
 
-  async exec(req: CreateDishRequest): Promise<Dish> {
-    const created = await this.dishModel.create(req);
-    await this.restaurantModel.findByIdAndUpdate(req.restaurant, { $push: { dishes: created } });
-    return plainToInstance(Dish, created);
+  async exec(req: CreateDishRequest): Promise<Dish | null> {
+    const restaurant = await this.restaurantModel.findById(req.restaurant);
+
+    if (restaurant?.isCompleted) {
+      const created = await this.dishModel.create(req);
+      await this.restaurantModel.findByIdAndUpdate(req.restaurant, { $push: { dishes: created } });
+
+      return plainToInstance(Dish, created);
+    } else throw new UnprocessableEntityException();
```

Comment:
Tutaj też bym zrobił mały refactor:
```suggestion
    const restaurant = await this.restaurantModel.findById(req.restaurant);
    if(!restuarant) return null;
    if(!restuarant.isCompleted) throw new UnprocessableEntityException('Profile not completed');
    
    const created = await this.dishModel.create(req);
    await this.restaurantModel.findByIdAndUpdate(req.restaurant, { $push: { dishes: created } });
    return plainToInstance(Dish, created);
```